nixos/modules/programs/gnupg.nix

```diff
@@ -135,7 +138,7 @@ in
       # The SSH agent protocol doesn't have support for changing TTYs; however we
       # can simulate this with the `exec` feature of openssh (see ssh_config(5))
       # that hooks a command to the shell currently running the ssh program.
-      Match host * exec "${cfg.package}/bin/gpg-connect-agent --quiet updatestartuptty /bye >/dev/null 2>&1"
```

Comment:
I would do
```
      Match host * exec "${runtimeShell} -c '${cfg.package}/bin/gpg-connect-agent --quiet updatestartuptty /bye >/dev/null 2>&1'"
```
so that we don't need to create a wrapper.

Reply:
Done with a slight modification: `'-c'`
because `exec ... -c` without quotes would fail in nushell, too.

Reply:
Actually, I'm not sure if either works. Calling `exec bash '-c' 'echo hello'` seems to be causing nushell 0.73.0 to crash (which is probably a nushell bug).

Calling `exec bash -c 'echo hello'` results in

```
Error: nu::parser::unknown_flag (link)

  × The `exec` command doesn't have flag `-c`.
   ╭─[entry #11:1:1]
 1 │ exec bash -c 'echo hello'
   ·            ┬
   ·            ╰── unknown flag
   ╰────
  help: Available flags: --help(-h). Use `--help` for more information.
```

Reply:
It looks like Nushell 0.74.0 will support passing flags to `exec` without quotes: https://github.com/nushell/nushell/pull/7527.

Reply:
Note that `exec` here is ssh_config's syntax, not the `exec` builtin command. I don't think this has nothing to do with the bug you linked. The shell just need to support the conventional calling syntax.

It would still be a better idea to use a POSIX shell rather than the user shell, but that's more of an openssh issue.